packages/desktop-client/src/components/manager/DeleteFile.tsx

```diff
@@ -1,35 +1,46 @@
 import React, { useState } from 'react';
 
+import { type File } from 'loot-core/src/types/file';
+
+import { type BoundActions } from '../../hooks/useActions';
 import { theme } from '../../style';
+import { type CommonModalProps } from '../../types/modals';
 import { ButtonWithLoading } from '../common/Button';
 import Modal from '../common/Modal';
 import Text from '../common/Text';
 import View from '../common/View';
 
-export default function DeleteMenu({ modalProps, actions, file }) {
-  const [loadingState, setLoadingState] = useState(null);
+type DeleteFileProps = {
+  modalProps: CommonModalProps;
+  actions: BoundActions;
+  file: File;
+};
 
-  async function onDeleteCloud() {
-    setLoadingState('cloud');
-    await actions.deleteBudget(file.id, file.cloudFileId);
-    setLoadingState(null);
+export default function DeleteFile({
+  modalProps,
+  actions,
+  file,
+}: DeleteFileProps) {
+  // If the state is "broken" that means it was created by another
+  // user. The current user should be able to delete the local file,
+  // but not the remote one
+  const isCloudFile = 'cloudFileId' in file && file.state !== 'broken';
 
-    modalProps.onBack();
-  }
+  const [loadingState, setLoadingState] = useState<'cloud' | 'local' | null>(
+    null,
+  );
 
-  async function onDeleteLocal() {
-    setLoadingState('local');
-    await actions.deleteBudget(file.id);
+  async function onDelete() {
+    setLoadingState(isCloudFile ? 'cloud' : 'local');
+    await actions.deleteBudget(
+      'id' in file ? file.id : undefined,
```

Comment:
Is it possible to use `file?.id` here?

Reply:
No because property `id` does not exist on `RemoteFile` type.